python/taichi/lang/impl.py

```diff
@@ -30,8 +30,11 @@ def expr_init(rhs):
 
 @taichi_scope
 def expr_init_list(xs, expected):
-    if not isinstance(xs, (list, tuple)):
+    import taichi as ti
+    if not isinstance(xs, (list, tuple, ti.Matrix)):
         raise TypeError(f'Cannot unpack type: {type(xs)}')
+    if isinstance(xs, ti.Matrix):
```

Comment:
Nit: I'd suggest to check `xs.m == 1` to make sure it's a vector.

Reply:
I think unpacking a matrix can also be useful?

Reply:
1. What's the typical use case for unpacking a matrix?
2. I believe it causes more confusion than usefulness, because the users will have to remember whether `ti.Matrix` is row-major or column-major.

Reply:
> Nit: I'd suggest to check `xs.m == 1` to make sure it's a vector.

I agree - unpacking a vector has a much clearer semantics then unpacking a matrix.